src/hotspot/cpu/aarch64/macroAssembler_aarch64.hpp

```diff
@@ -437,11 +437,14 @@ class MacroAssembler: public Assembler {
     Assembler::INSN(Rd, Rn, Rm, Ra);                                          \
   }
 
-  WRAP(madd) WRAP(msub) WRAP(maddw) WRAP(msubw)
+  WRAP(madd) WRAP(maddw)
   WRAP(smaddl) WRAP(smsubl) WRAP(umaddl) WRAP(umsubl)
 #undef WRAP
 
 
+  void msub(Register Rd, Register Rn, Register Rm, Register Ra, Register tmp = rscratch2);
+  void msubw(Register Rd, Register Rn, Register Rm, Register Ra, Register tmp = rscratch2);
```

Comment:
Please delete these two methods that use rscratch2 as a default tmp register.

Reply:
I tried it and encountered a problem: msub/msubw will be called in other functions,

  inline void mnegw(Register Rd, Register Rn, Register Rm) {
    msubw(Rd, Rn, Rm, zr);
  }

  inline void mneg(Register Rd, Register Rn, Register Rm) {
    msub(Rd, Rn, Rm, zr);
  }

If we add a parameter to msub/msubw, then

1. All these functions (mnegw, mneg, ...) that call msub/msubw also need to add parameters,

2. Moreover, all the calling functions of mnegw and mneg also need to be modified.

The above two effects involve too many code changes. 

Please see if we can keep a default parameter for msub/msubw so that their calling functions do not need to be modified.

Reply:
A hidden clobber of a scratch register makes msub dangerous to use, as the failure of the first version of this PR proved. You don't have to convert 100% of the uses of `msub`, just those that are frequently executed.